libs/common-system/src/main/java/org/codice/ddf/configuration/DictionaryMap.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.configuration;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A HashMap based Dictionary for use with OSGI methods requiring a Dictionary
+ *
+ * @param <K>
+ * @param <V>
+ */
+public class DictionaryMap<K, V> extends Dictionary<K, V> implements Map<K, V>, Serializable {
```

Comment:
Seeing how we extensively seemed to be creating dictionaries left and right when sometimes we are doing so only to add one or 2 entries to it. Should we add a few static methods like these and simplify the code elsewhere? Kinda of what is available in ImmutableMap.

```of(K k, Vv)
of (K k1, V v1, K k2, V v2)
of(K k1, V v1, K k2, V v2, K k3, V v3)
of(K k1, V v1, K k2, V v2, K k3, V v3, K k4, V v4)
of(K k1, V v1, K k2, V v2, K k3, V v3, K k4, V v4, K k5, V v5)```

Reply:
Added the methods but will leave the current usage as is.